### PR TITLE
enhancement/13065-boxplot-dashStyle

### PR DESCRIPTION
**Pull request by raf18seb**
Added individual dash style settings for various parts of box plots. See #13065. 

~~Enhancement #13065 , added more ways for styling dash styles in the boxplot series.~~

### CODE REVIEW

File: samples/unit-tests/series-boxplot/boxplot/demo.js [hunk 1]
```diff
@@ -190,4 +190,79 @@ QUnit.test(
...
+
+        var series = chart.series[0],
+            firstPoint = series.points[0],
+            secondPoint = series.points[1];
+
+        ['box', 'medianShape', 'stem', 'whiskers'].forEach(function (elem) {
+            assert.strictEqual(
+                firstPoint[elem].attr('stroke-dasharray'),
+                '1,3',
+                'Point\'s ' + elem + 'DashStyle'
```
Comment (pawelfus) [inline samples/unit-tests/series-boxplot/boxplot/demo.js]: Friendly reminder for this an other tests: make this a full sentence, e.g. `...dashStyle should be applied to a point`.

File: ts/parts-more/BoxPlotSeries.ts [hunk 2]
```diff
@@ -464,8 +514,11 @@ seriesType<Highcharts.BoxPlotSeries>('boxplot', 'column', {
                         options.stemWidth,
                         options.lineWidth
                     );
-                    stemAttr.dashstyle =
-                        point.stemDashStyle || options.stemDashStyle;
+                    stemAttr.dashstyle = pick(
+                        point.stemDashStyle,
+                        options.stemDashStyle,
+                        options.dashStyle
+                    );
```
Comment (pawelfus) [inline ts/parts-more/BoxPlotSeries.ts]: Not a problem, just a hint: in general `pick()` is useful when dealing with numerical values, where `a || b` can not be used (as `a=0` would pick `b` value).

File: ts/parts-more/BoxPlotSeries.ts [hunk 3]
```diff
@@ -723,4 +788,64 @@ seriesType<Highcharts.BoxPlotSeries>('boxplot', 'column', {
...
+ * @sample {highcharts} highcharts/css/boxplot/
+ *         Box plot in styled mode
+ *
+ * @type      {Highcharts.DashStyleValue}
+ * @default   Solid
+ * @since     next
+ * @product   highcharts
+ * @apioption series.boxplot.data.boxDashStyle
+ */
+
```
Comment (pawelfus) [inline ts/parts-more/BoxPlotSeries.ts]: Do need to define twice all options for docs?

Comment (raf18seb) [reply ts/parts-more/BoxPlotSeries.ts]: I think yes is we want to show them in API not only on the series level but on the point level too: https://api.highcharts.com/highcharts/series.boxplot.data
What is the better practice? Should I remove them from `boxplot.data` level?

Comment (pawelfus) [reply ts/parts-more/BoxPlotSeries.ts]: No, my bad, didn't realize it's `boxplot.data`, not `boxplot`. Thanks!

**Pull request merged**